static/downloadables/grep_file.py

```diff
@@ -258,56 +263,33 @@ def isSnpInFilters(rsID, chromPos, tableObjDict, isAllFiltersNone, traits, study
+# assumes the file is valid (validated with getZippedFileExtension function)
+def openFileForParsing(inputFile):
     filename = ""
-    compressed = False
-    # if the file is zipped
     if zipfile.is_zipfile(inputFile):
         # open the file
         archive = zipfile.ZipFile(inputFile)
-        validArchive = []
         # get the vcf or txt file in the zip
         for filename in archive.namelist():
-            if filename[-4:].lower() == ".vcf" or filename[-4:].lower() == ".txt":
-                validArchive.append(filename)
-                filename = validArchive[0]
-
-    # TODO: Figure this out: if the file is tar zipped
+            extension = filename[-4:].lower()
+            if extension == ".txt" or extension == ".vcf":
+                # TextIOWrapper converts bytes to strings and force_zip64 is for files potentially larger than 2GB
+                return TextIOWrapper(archive.open(filename, force_zip64=True))
     elif tarfile.is_tarfile(inputFile):
         # open the file
         archive = tarfile.open(inputFile)
         # get the vcf or txt file in the tar
         for tarInfo in archive:
-            if tarInfo.name[-4:].lower() == ".vcf" or tarInfo.name[-4:].lower() == ".txt":
-                # wrap the ExFileObject in an io.TextIOWrapper and read
-                if tarInfo.name[-4:].lower() == ".txt":
-                    newFile = archive.extractfile(tarInfo) 
-                    return newFile
-                else:
-                    newFile = archive.extractfile(tarInfo)
-                    vcf_reader = vcf.Reader(newFile)
-                    return vcf_reader
-                new_file = io.TextIOWrapper(archive.extractfile(tarInfo)).read().replace("\r", "").split("\n")
-    # if file is gzipped
+            extension = tarInfo.name[-4:].lower()
+            if extension == ".txt" or extension == ".vcf":
+                # TextIOWrapper converts bytes to strings
+                return TextIOWrapper(archive.extractfile(tarInfo))
     elif inputFile.lower().endswith(".gz") or inputFile.lower().endswith(".gzip"):
-        filename = inputFile
-        compressed= True
-    else:
-        # default open for regular vcf and txt files
-        filename = inputFile
-        compressed = False
-    # return the new file either as is (txt) or as a vcf.Reader (vcf)
-    if isTxtExtension:
-        txt_file = open(filename, 'r')
-        return txt_file
+        return TextIOWrapper(gzip.open(inputFile, 'r'))
     else:
-        # open the newly unzipped file using the vcf reader
-        try:
-            vcf_reader = vcf.Reader(filename = filename, compressed = compressed)
-            return vcf_reader
-        except:
-            # typically happens if the file is empty
-            raise SystemExit("The VCF file is not formatted correctly. Each line must have 'GT' (genotype) formatting and a non-Null value for the chromosome and position.")
+        # default option for regular vcf and txt files
```

Comment:
Do we use the PyVCF module anymore? Do we use it to write to files in the calculate python script? If not, we should make sure to remove that from the bash script. Also, can you check the import statements at the top of these python  scripts and remove ones we don't use